src/client/sandbox/event/index.js

```diff
@@ -1,210 +1,235 @@
-
-        this.listeners             = Listeners;             // TODO:
-        this.unload                = new Unload(sandbox);
-        this.selection             = Selection;             // TODO:
-        this.eventSimulator        = EventSimulator;        // TODO:
-        this.elementEditingWatcher = ElementEditingWatcher; // TODO:
-        this.focusBlur             = FocusBlur;             // TODO:
-    }
-
-    //wrappers
-    _overridedDispatchEvent (ev) {
-        Listeners.beforeDispatchEvent();
-
-        var res = NativeMethods.dispatchEvent.call(this, ev);
-
-        Listeners.afterDispatchEvent();
-
-        return res;
-    }
-
-    _overridedFireEvent (eventName, ev) {
-        var eventType = eventName.substring(0, 2) === 'on' ? eventName.substring(2) : eventName;
-        var createEventType;
-        var res;
-
-        Listeners.beforeDispatchEvent();
-
-        //event is 'MSEventObj'
-        if (!ev || !ev.target) {
-            if (/(^mouse\w+$)|^(dbl)?click$|^contextmenu$/.test(eventType))
-                createEventType = 'MouseEvents';
-            else if (/^key\w+$/.test(eventType))
-                createEventType = 'Events';
-            else if (/^touch\w+$/.test(eventType))
-                createEventType = 'TouchEvent';
-            else
-                createEventType = 'Events';
-
-            if (ev) {
-                ev = extend(document.createEvent(createEventType), ev);
-                ev.initEvent(eventType, typeof ev.cancelBubble !== 'undefined' ? ev.cancelBubble : false, true);
+        this.listeners             = new Listeners();
+        this.unload                = new UnloadSandbox(sandbox);
+        this.timers                = new TimersSandbox(sandbox);
+        this.eventSimulator        = new EventSimulator();
+        this.selection             = new Selection(this);
+        this.elementEditingWatcher = new ElementEditingWatcher(this.eventSimulator);
+        this.focusBlur             = new FocusBlurSandbox(sandbox, this.eventSimulator);
+
+        this.overridedDispatchEvent     = null;
+        this.overridedFireEvent         = null;
+        this.overridedAttachEvent       = null;
+        this.overridedDetachEvent       = null;
+        this.overridedClick             = null;
+        this.overridedSetSelectionRange = null;
+        this.overridedSelect            = null;
+        this.overridedFocus             = null;
+        this.overridedBlur              = null;
+
+        this.onFocus              = null;
+        this.cancelInternalEvents = null;
+
+        this._createOverridedMethods();
+        this._createInternalHandlers();
+    }
+
+    _createOverridedMethods () {
+        var selection        = this.selection;
+        var focusBlurSandbox = this.focusBlur;
+        var listeners        = this.listeners;
+        var eventSimulator   = this.eventSimulator;
+
+        this.overridedDispatchEvent = function (ev) {
```

Comment:
It would be nice to have these methods compound into the object, e.g.: `overridedMethods.dispatchEvent`.
It will give us 2 benefits:
1)We will get rid of `overrided` prefix.
2)`this` inside them will not be so confusing. Currently it's intended to the element, but from the code it's looks like `this` points to `EventSandbox`